constructr-machine/src/main/scala/de/heikoseeberger/constructr/machine/ConstructrMachine.scala

```diff
@@ -158,6 +162,10 @@ final class ConstructrMachine[A: Coordination.AddressSerialization, B <: Coordin
 
   // Refreshing
 
+  onTransition {
+    case _ -> State.Refreshing => coordination.refresh(selfAddress, addOrRefreshTtl, stateData.context).pipeTo(self)
+  }
+
   when(State.Refreshing, coordinationTimeout) {
     case Event(Coordination.Refreshed(_), _) => goto(State.RefreshScheduled).using(stateData.copy(coordinationRetriesLeft = coordinationRetries))
   }
```

Comment:
Oops, was that really missing? Thanks for catching! We need ways better tests to avoid such mistakes.